_pytest/python.py

```diff
@@ -1106,28 +1125,20 @@ def _idval(val, argname, idx, idfn):
         try:
             s = idfn(val)
             if s:
-                return s
+                return _escape_strings(s)
         except Exception:
             pass
 
-    if isinstance(val, bytes):
-        return _escape_bytes(val)
+    if isinstance(val, bytes) or (_PY2 and isinstance(val, unicode)):
+        return _escape_strings(val)
     elif isinstance(val, (float, int, str, bool, NoneType)):
         return str(val)
     elif isinstance(val, REGEX_TYPE):
-        return _escape_bytes(val.pattern) if isinstance(val.pattern, bytes) else val.pattern
+        return _escape_strings(val.pattern) if isinstance(val.pattern, bytes) else val.pattern
```

Comment:
given the new idea for unicode, i feel we no longer need the conditional, please check if thats the case